src/DataTypes/DataTypeSet.h

```diff
@@ -20,7 +20,7 @@ class DataTypeSet final : public IDataTypeDummy
     bool isParametric() const override { return true; }
 
     // Used for expressions analysis.
-    MutableColumnPtr createColumn() const override { return ColumnSet::create(0, nullptr); }
+    MutableColumnPtr createColumn() const override { return ColumnSet::create(0, FutureSet{}); }
```

Comment:
Hmm.. Not sure about this. Previously there was a nullptr there, which was an unusable set. I guess if we make it a valid empty Set now it might allow some unexpected erroneous usages.

Reply:
Got it, in such case current version corresponds to the current behavior.